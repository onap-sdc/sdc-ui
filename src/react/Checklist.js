import React from 'react';
import PropTypes from 'prop-types';
import Checkbox from './Checkbox.js';

const Checklist = ({ items = [], className, onChange }) => (
    <div className={className}>
        {items.map((item, index) => {
            return (
                <div key={`checkbox-item-${index}`} className="checkbox-item">
                    <Checkbox
                        key={`${item.label}${index}`}
                        label={item.label}
                        value={item.value}
                        checked={item.checked}
                        disabled={item.disabled}
                        onChange={value => {
                            let obj = {};
                            obj[item.value] = value;
                            onChange(obj);
                        }}
                        data-test-id={item.dataTestId}
                    />
                </div>
            );
        })}
    </div>
);

Checklist.propTypes = {
    items: PropTypes.arrayOf(
        PropTypes.shape({
            label: PropTypes.string,
            value: PropTypes.string,
            checked: PropTypes.bool,
            disabled: PropTypes.bool,
            dataTestId: PropTypes.string
        })
    ),
    className: PropTypes.string,
    onChange: PropTypes.func
};

export default Checklist;
