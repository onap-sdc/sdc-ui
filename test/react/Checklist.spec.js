import React from 'react';
import Checklist from '../../src/react/Checklist.js';

import renderer from 'react-test-renderer';

const items = [
    {
        label: 'apple',
        value: 'apple',
        dataTestId: 'apple',
        checked: true
    },
    {
        label: 'banana',
        value: 'banana',
        dataTestId: 'banana',
        checked: false
    },
    {
        label: 'orange',
        value: 'orange',
        dataTestId: 'orange',
        checked: true
    }
];

const itemsDisabled = [
    {
        label: 'apple',
        value: 'apple',
        dataTestId: 'apple',
        checked: true,
        disabled: true
    },
    {
        label: 'banana',
        value: 'banana',
        dataTestId: 'banana',
        checked: false,
        disabled: true
    },
    {
        label: 'orange',
        value: 'orange',
        dataTestId: 'orange',
        checked: false
    }
];


describe('Checklist', () => {
    test('Checklist - Default ', () => {
        const checklist = renderer.create(<Checklist items={items} onChange={() => { }} />).toJSON();
        expect(checklist).toMatchSnapshot();
    });

    test('Checklist - With disabled items ', () => {
        const checklist = renderer.create(<Checklist items={itemsDisabled} onChange={() => { }} />).toJSON();
        expect(checklist).toMatchSnapshot();
    });



});