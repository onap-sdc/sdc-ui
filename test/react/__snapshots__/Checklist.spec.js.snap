// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checklist Checklist - Default  1`] = `
<div
  className={undefined}
>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={true}
          className="sdc-checkbox__input"
          data-test-id="apple"
          disabled={undefined}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="apple"
        />
        <span
          className="sdc-checkbox__label"
        >
          apple
        </span>
      </label>
    </div>
  </div>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={false}
          className="sdc-checkbox__input"
          data-test-id="banana"
          disabled={undefined}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="banana"
        />
        <span
          className="sdc-checkbox__label"
        >
          banana
        </span>
      </label>
    </div>
  </div>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={true}
          className="sdc-checkbox__input"
          data-test-id="orange"
          disabled={undefined}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="orange"
        />
        <span
          className="sdc-checkbox__label"
        >
          orange
        </span>
      </label>
    </div>
  </div>
</div>
`;

exports[`Checklist Checklist - With disabled items  1`] = `
<div
  className={undefined}
>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={true}
          className="sdc-checkbox__input"
          data-test-id="apple"
          disabled={true}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="apple"
        />
        <span
          className="sdc-checkbox__label"
        >
          apple
        </span>
      </label>
    </div>
  </div>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={false}
          className="sdc-checkbox__input"
          data-test-id="banana"
          disabled={true}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="banana"
        />
        <span
          className="sdc-checkbox__label"
        >
          banana
        </span>
      </label>
    </div>
  </div>
  <div
    className="checkbox-item"
  >
    <div
      className="sdc-checkbox "
    >
      <label>
        <input
          checked={false}
          className="sdc-checkbox__input"
          data-test-id="orange"
          disabled={undefined}
          name={undefined}
          onChange={[Function]}
          type="checkbox"
          value="orange"
        />
        <span
          className="sdc-checkbox__label"
        >
          orange
        </span>
      </label>
    </div>
  </div>
</div>
`;
